.pylintrc

```diff
@@ -7,6 +7,9 @@ function-naming-style=camelCase
 argument-naming-style=camelCase
 variable-naming-style=camelCase
 
+[DESIGN]
+max-attributes=15
+
 [MESSAGES CONTROL]
 disable-checker=wrong-hanging-indentation,invalid-name,indent-after-paren,bad-continuation
-disable=bad-continuation,bad-whitespace,import-error,ungrouped-imports,missing-docstring,invalid-name,multiple-imports,unidiomatic-typecheck,line-too-long,unused-variable,too-few-public-methods,bare-except,trailing-whitespace,too-many-locals,too-many-branches,too-many-statements,unused-argument,fixme,redefined-builtin,consider-using-ternary,no-else-return,superfluous-parens,too-many-arguments,old-style-class,bad-indentation,wrong-import-order,unused-import,logging-not-lazy,anomalous-backslash-in-string,no-self-use,trailing-newlines,expression-not-assigned,pointless-statement,inconsistent-return-statements,attribute-defined-outside-init,broad-except, deprecated-lambda, duplicate-code, len-as-condition, multiple-statements, no-init, no-value-for-parameter, protected-access,redefined-outer-name, singleton-comparison,super-init-not-called,too-many-lines,too-many-nested-blocks,too-many-return-statements,unneeded-not,too-many-ancestors,deprecated-method,useless-import-alias,useless-object-inheritance,consider-using-set-comprehension,undefined-variable,useless-import-alias,too-many-ancestors,consider-using-dict-comprehension,invalid-envvar-default
+disable=bad-continuation,bad-whitespace,import-error,ungrouped-imports,missing-docstring,invalid-name,multiple-imports,unidiomatic-typecheck,line-too-long,unused-variable,too-few-public-methods,bare-except,trailing-whitespace,too-many-locals,too-many-branches,too-many-statements,unused-argument,fixme,redefined-builtin,consider-using-ternary,no-else-return,superfluous-parens,too-many-arguments,old-style-class,bad-indentation,wrong-import-order,unused-import,logging-not-lazy,anomalous-backslash-in-string,no-self-use,trailing-newlines,expression-not-assigned,pointless-statement,inconsistent-return-statements,attribute-defined-outside-init,broad-except, deprecated-lambda, duplicate-code, len-as-condition, multiple-statements, no-init, no-value-for-parameter, protected-access,redefined-outer-name, singleton-comparison,super-init-not-called,too-many-lines,too-many-nested-blocks,too-many-return-statements,unneeded-not,too-many-ancestors,deprecated-method,useless-import-alias,useless-object-inheritance,consider-using-set-comprehension,undefined-variable,useless-import-alias,too-many-ancestors,consider-using-dict-comprehension,invalid-envvar-default,logging-format-interpolation
```

Comment:
Is the masking due to changes of this patch, or changes in pylint (i.e. is it a new feature?). If the latter, can we keep it clean? The motivation (https://docs.python.org/3/howto/logging.html#optimization) seems to be reasonable (in particular for the `debug` log level).

Reply:
It's a matter of policy, I can change it. I personally find that using `%s`, etc. less readable - I am aware of the motivation but in my case there was nothing performance critical. Let's say, I have favoured readability over performance. But again, your call - I can reformat the messages with `%s` and remove the linter exception.

(Maybe, Python 3's [f-strings](https://realpython.com/python-f-strings/) will address both the performance and readability issue?)

Reply:
If this is the only patch that triggers it and this is the only place were we use the other syntax, yes, I think we should stay consistent with the rest and we should not have yet another linter exception.

We can of course decide that we want to go the other way around, but then we should do so everywhere, not in one single place.

Reply:
Fully agreed :+1: